cmd/podman/ps.go

```diff
@@ -288,7 +300,77 @@ func psCmd(c *cli.Context) error {
 		outputContainers = []*libpod.Container{latestCtr}
 	}
 
-	return generatePsOutput(outputContainers, opts)
+	pss := shared.PBatch(outputContainers, 8, opts)
+	if opts.Sort != "" {
+		pss, err = sortPsOutput(opts.Sort, pss)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If quiet, print only cids and return
+	if opts.Quiet {
+		return printQuiet(pss, opts)
+	}
+
+	if opts.Format != "" {
+		if opts.Format == "json" {
+			return dumpJSON(pss)
+		}
+		return printFormat(opts.Format, pss, opts)
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	if !opts.Namespace {
+		fmt.Fprintf(w, "\n%s\t%s\t%s\t%s\t%s\t%s\t%s", hid, himage, hcommand, hcreated, hstatus, hports, hnames)
+		if !opts.Size && !opts.Pod {
+			fmt.Fprintf(w, "\t%s", hinfra)
+		}
+		if opts.Pod {
+			fmt.Fprintf(w, "\t%s", hpod)
+		}
+		if opts.Size {
+			fmt.Fprintf(w, "\t%s", hsize)
+		}
+
+	} else {
+		fmt.Fprintf(w, "\n%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s", hid, hnames, nspid, nscgroup, nsipc, nsmnt, nsnet, nspidns, nsuserns, nsuts)
+	}
+
+	defer w.Flush()
```

Comment:
Curious about why your defer is down here?

Reply:
no reason, to tighten it up, i moved it next to where the writer is defined. thanks